llvm/cmake/modules/HandleLLVMOptions.cmake

```diff
@@ -289,6 +289,39 @@ function(has_msvc_incremental_no_flag flags incr_no_flag_on)
   endif()
 endfunction()
 
+macro(add_compile_option_ext flag name)
```

Comment:
If this patch is not supposed to be upstreamed to llvm-project, please, move new functionality to a separate file to avoid merge conflicts with community changes.